webview/src/experiments/components/table/tableDataSlice.ts

```diff
@@ -0,0 +1,45 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { TableData } from 'dvc/src/experiments/webview/contract'
+
+export interface TableDataState extends TableData {
+  hasData?: boolean
+}
+
+export const tableDataInitialState: TableDataState = {
+  changes: [],
+  columnOrder: [],
+  columnWidths: {},
+  columns: [],
+  filteredCounts: {
+    checkpoints: 0,
+    experiments: 0
+  },
+  filters: [],
+  hasCheckpoints: false,
+  hasColumns: false,
+  hasData: false,
+  hasRunningExperiment: false,
+  rows: [],
+  sorts: []
+}
+
+export const tableDataSlice = createSlice({
+  initialState: tableDataInitialState,
+  name: 'tableData',
+  reducers: {
+    update: (state, action: PayloadAction<TableData>) => {
+      if (action.payload) {
+        return {
+          ...state,
+          ...action.payload,
+          hasData: true
+        }
```

Comment:
This is pretty much the same as it was before. An improvement here would be to go through previous columns, rows and objects to change the reference of only what changed. I will do that as a follow up.